src/vm/mlinfo.cpp

```diff
@@ -2796,9 +2796,18 @@ MarshalInfo::MarshalInfo(Module* pModule,
                 }
 #endif // FEATURE_COMINTEROP
 
-                if (m_pMT->HasInstantiation())
+                if (m_pMT->HasInstantiation() && (!m_pMT->IsBlittable()
+                    || m_pMT->HasSameTypeDefAs(g_pByReferenceClass)
+                    || m_pMT->HasSameTypeDefAs(g_pNullableClass)
+                    || m_pMT->HasSameTypeDefAs(MscorlibBinder::GetClass(CLASS__VECTOR64T))
+                    || m_pMT->HasSameTypeDefAs(MscorlibBinder::GetClass(CLASS__VECTOR128T))
+                    || m_pMT->HasSameTypeDefAs(MscorlibBinder::GetClass(CLASS__VECTOR256T))
+#ifndef CROSSGEN_COMPILE
```

Comment:
Why is this ifndef necessary? We try to avoid logic differences like this for crossgen.

Reply:
`Vector<T>` is blocked from even being type loaded during crossgen (due to its variable size): https://github.com/dotnet/coreclr/blob/master/src/vm/methodtablebuilder.cpp#L1182

Reply:
Ok, make sense. I have not realized that this is to avoid failing here 100% of time during crossgen. Comment may be nice

Reply:
Is there a matching change to be done for this in crossgen2?

Reply:
I'm not sure. Looking at crossgen2, I don't see any specific handling of `Vector<T>` and disallowing it as a type.

I did find a section for the FieldLayoutAlgorithm: https://github.com/dotnet/coreclr/blob/master/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunCompilerContext.cs#L50
However, that field layout only supports `Vector64/128/256<T>`: https://github.com/dotnet/coreclr/blob/master/src/tools/crossgen2/Common/Compiler/VectorFieldLayoutAlgorithm.cs

The only logic I can find for `Vector<T>` is in the `SystemVStructClassificator` where it doesn't treat it as an EightBytes struct: https://github.com/dotnet/coreclr/blob/master/src/tools/crossgen2/Common/JitInterface/SystemVStructClassificator.cs#L273

Reply:
We may need a change here: https://github.com/dotnet/coreclr/blob/master/src/tools/crossgen2/Common/TypeSystem/Interop/IL/MarshalHelpers.cs#L337

cc @fadimounir

Reply:
What is the easiest way to test Crossgen2 on an specific test?

Reply:
I've pushed up a commit which I believe handles this, but it's not been tested (and doesn't look to be covered by CI currently).